devito/builtins.py

```diff
@@ -53,6 +54,135 @@ def smooth(f, g, axis=None):
         dv.Operator(dv.Eq(f, g.avg(dims=axis)), name='smoother')()
 
 
+def gaussian_smooth(f, sigma=1, _order=4, mode='reflect'):
+    """
+    Gaussian smooth function.
+    """
+    class ObjectiveDomain(dv.SubDomain):
+
+        name = 'objective_domain'
+
+        def __init__(self, lw):
+            super(ObjectiveDomain, self).__init__()
+            self.lw = lw
+
+        def define(self, dimensions):
+            return {d: ('middle', self.lw, self.lw) for d in dimensions}
+
+    def smooth(fo, fi, data):
+        slices = tuple([slice(lw, -lw) for _ in range(fi.grid.dim)])
+        fi.data[slices] = data
+        eqs = []
+        for d in fi.dimensions:
+            dim_l = dv.SubDimension.left(name='abc_%s_l' % d.name, parent=d,
+                                         thickness=lw)
+            dim_r = dv.SubDimension.right(name='abc_%s_r' % d.name, parent=d,
+                                          thickness=lw)
+            if mode == 'constant':
```

Comment:
aside from the `to_copy` variables, the construction of the `eqs` is pretty much identical in the two cases (constant, reflect). So we can probably trim down the bodies of this if-else ?

Reply:
Yes, can do.